yatta/src/main.rs

```diff
@@ -630,6 +625,20 @@ fn handle_socket_message(
                                 float_titles.push(target)
                             }
                         }
+                        SocketMessage::SetWorkspace(index) => {
+                            d.set_workspace(index);
+                        }
+                        SocketMessage::MoveWindowToWorkspace(index) => {
```

Comment:
I would like to have an accompanying `MoveWindowToWorkspaceAndFollow` message since this is what "feels" like the natural behaviour to me. The handler would be the same as here with the addition of this at the end:

```rust
d.set_workspace(index);
```

For the correct window to be focused when following a moved window to its new workspace, we would need to modify the `WindowsEventType::Show` handler in `main.rs` to explicitly set the window emitting the show event to be the new foreground window:

```rust
let idx = display.get_foreground_window_index() + 1;
// If we are inserting where there is a window that has resize adjustments, take
// over those resize adjustments and remove them from the window that is
// currently there
if let Some(current_window) = display.get_current_windows_mut().get_mut(idx) {
    let resize = current_window.resize.clone();
    current_window.resize = None;
    ev.window.resize = resize;
}

display.get_current_windows_mut().insert(idx, ev.window);
display.calculate_layout();
display.apply_layout(None);
ev.window.set_foreground();

if let Some(title) = ev.window.title() {
    if let Ok(path) = ev.window.exe_path() {
        info!(
            "managing new window: {} - {} ({})",
            &exe_name_from_path(&path),
            &title,
            ev.window.hwnd.0
        );
    }
}
```

Reply:
This seems like a great idea! Not something that I would use, as I move windows simply to remove then when they are cluttering my current workspace, but I'll implement this right now.